types.go

```diff
@@ -150,3 +150,21 @@ type Capabilities interface {
 	// Tagging returns whether the reporter has the capability for tagged metrics.
 	Tagging() bool
 }
+
+// SuccessFilterFn determines whether an error should be considered success
+type SuccessFilterFn func(err error) bool
+
+// ExecFn will be executed in an InstrumentedCall
+type ExecFn func() error
+
+// An InstrumentedCall allows tracking the success, errors, and timing of blocks of code
+type InstrumentedCall interface {
+	// Exec executes the given block of code, and records whether it succeeded or
+	// failed, and the amount of time that it took
+	Exec(f ExecFn) error
+
+	// ExecWithFilter executes the given block of code, and records whether it succeeded or
+	// failed based on the result of a custom filter (e.g. the filter could determine a bad request error
+	// to be actually success for server logic), and the amount of time that it took
+	ExecWithFilter(f ExecFn, isSuccess SuccessFilterFn) error
```

Comment:
I'm not sure I follow the value of the additional method here. If needed, wouldn't it be as simple as a lambda wrapper around the `ExecFn`?

i.e. lets say you have some code using `ExecWithFilter` like so –
```
func execFn() error {
  // ... your exec fn
}

func successFilterFn(error) bool {
  // ... your filter fn
}


func XYZ() {
  instrumentedCall := ...
  instrumentedCall.ExecWithFilter(execFn, successFilterFn)
}
```

Isn't that the same as –

```
// other func defs 
func XYZ {
  instrumentedCall := ...
  instrumentedCall.Exec(func() error {
  err := execFn()
    if !successFilterFn(err) {
      return err
    }
  return nil
  })
}
```

Reply:
I wanted to keep the interface the same as the go-common interface so when migrating you can use the same functions.

Reply:
You would have to use the new constructor either way however.  It does seem like not a fantastic API when the way @prateek pointed out is actually simpler and provides better readability in terms of users who don't know what the second lambda does exactly without reading the interface.